synapse/storage/databases/main/receipts.py

```diff
@@ -354,7 +356,7 @@ def f(txn: LoggingTransaction) -> List[Dict[str, Any]]:
     )
     async def _get_linearized_receipts_for_rooms(
         self, room_ids: Collection[str], to_key: int, from_key: Optional[int] = None
-    ) -> Dict[str, List[JsonDict]]:
+    ) -> Dict[str, Sequence[JsonDict]]:
```

Comment:
I think this `Dict` is okay, because it comes from `@cachedList`?

Reply:
Yes, `cachedList` doesn't do the caching. 👍 The values in the dict need to be non-mutable.